server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -2952,7 +2952,7 @@ public void testRefreshListenersDuringPeerRecovery() throws IOException {
             });
 
             PlainActionFuture<Void> listener = PlainActionFuture.newFuture();
-            shard.addRefreshListener(10, listener);
+            shard.addRefreshListener(10, false, listener);
```

Comment:
I think we can use `randomBoolean()` instead of false? Also in the next two changes.

```suggestion
            shard.addRefreshListener(10, randomBoolean(), listener);
```